rholang/src/test/scala/BlockRandomSeedSpec.scala

```diff
@@ -0,0 +1,62 @@
+import cats.syntax.all._
+import coop.rchain.crypto.PublicKey
+import coop.rchain.rspace.hashing.Blake2b256Hash
+import coop.rchain.shared.Base16
+import coop.rchain.store.InMemoryKeyValueStore
+import monix.eval.Task
+import monix.execution.Scheduler.Implicits.global
+import org.scalacheck.Prop.{forAll, AnyOperators}
+import org.scalacheck.{Arbitrary, Gen}
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatestplus.scalacheck.Checkers.check
+import scodec.bits.ByteVector
+
+class BlockRandomSeedSpec extends AnyFlatSpec {
+  "encoding BlockRandomSeed object and decoding it" should "give original object" in {
+    val blockRandomSeedGen = for {
+      shardId                <- Gen.stringOfN(6, Gen.alphaChar)
+      blockNumber            <- Gen.choose(0L, 1000000L)
+      senderList             <- Gen.listOfN(120, Gen.hexChar)
+      sender                 = PublicKey(senderList.map(_.toByte).toArray)
+      validatorPublicKeyList <- Gen.listOfN(120, Gen.hexChar)
+      validatorPublicKey     = PublicKey(validatorPublicKeyList.map(_.toByte).toArray)
+      hash                   <- Gen.listOfN(Blake2b256Hash.length, Gen.hexChar)
+      preStateHash           = Blake2b256Hash.fromByteArray(hash.map(_.toByte).toArray)
+
+      _ = println(s"")
+      _ = println(s"sender: ${ByteVector(sender.bytes)}")
+      _ = println(s"validatorPublicKey: ${ByteVector(validatorPublicKey.bytes)}")
+    } yield BlockRandomSeed(shardId, blockNumber, sender, validatorPublicKey, preStateHash)
+
+    implicit lazy val blockRandomSeedArbitrary = Arbitrary(blockRandomSeedGen)
+    check {
+      forAll { blockRandomSeed: BlockRandomSeed =>
+        val serialized = BlockRandomSeed.codecBlockRandomSeed.encode(blockRandomSeed).require
+
+        println(s"source object: ${blockRandomSeed.toString}")
+        println(s"serialized: ${serialized.toByteVector}")
+
+        val reconstructed = BlockRandomSeed.codecBlockRandomSeed.decode(serialized).require.value
+
+        reconstructed ?= blockRandomSeed
+      }
+    }
+  }
+
+  "encoding Blake2b256Hash object and decoding it" should "give original object" in {
+    val hashGen = for {
+      hashValue <- Gen.listOfN(Blake2b256Hash.length, Gen.hexChar)
+    } yield Blake2b256Hash.fromByteArray(hashValue.map(_.toByte).toArray)
+
+    implicit lazy val blockRandomSeedArbitrary = Arbitrary(hashGen)
+
+    check {
+      forAll { hash: Blake2b256Hash =>
+        val serialized = Blake2b256Hash.codecPureBlake2b256Hash.encode(hash).require
+        val restored   = Blake2b256Hash.codecPureBlake2b256Hash.decode(serialized).require.value
+
+        restored ?= hash
+      }
+    }
```

Comment:
Please add another test with constants for each field and check that result is expected random seed. The purpose is to detect any change in serializer of random generator.